### ISSUE TITLE
Feature Request: Save Workflow Hash

### ISSUE THREAD

**Issue by BeTheRobotSD** [alexopus/ComfyUI-Image-Saver#35]
CivitAI does not currently provide a clear way to link back to the workflow that generated your image. As a workflow developer, it would be nice to have the images generated using my workflows to show on the workflow page just like how they do on the Model pages. I'm not sure what the best method would be to support this, but I am proposing a few options:

1. hash_input: string - This would support adding any single hash to the output, which could support pre-defining  the hash for the workflow in a node and using that as input for this field.
2. hash_inputs: list - This would support adding multiple hash inputs, covering future usecases where there may be multiple hashes you want to add beyond just the workflow hash.

Neither of these are foolproof, as someone can put whatever they want into this field. I don't necessarily see this as a blocker though, as I can already add any LoRA / Embedding at strength 0 to the prompt and have it link to that without any effect from it.

Thanks for reviewing my suggestion!

**Comment by alexopus** [comment]
Good idea, but is it supported by CivitAI? Can you upload an image with a workflow hash that will get linked?
If yes, please provide examples.

**Comment by PrometheusDante** [comment]
That would be nice indeed. Having to post all showcase images directly under the workflow is not a problem, but no one using the resource usually does, so it always looks rather empty and unused.
And as workflows have hash and AIR displayed on the download page, it should be useable as an identifier somehow.

Depending on how CivitAI is reading those values out it might just be a matter of appending that to an image's meta, but I did not try tinkering with it like that yet

**Comment by diodiogod** [comment]
I agree with this, it would be awesome to be able to attach a manual hash to your workflow, even if it's a previous version workflow.

**Comment by alexopus** [comment]
This is now available in v1.5.0.
You can enter up to comma-separated Auto3 hashes in the additional_hashes input and they will be stored with the image.

Thanks to @diodiogod for the impl!

**Comment by diodiogod** [comment]
And just to explain thing a little bit further. @BeTheRobotSD you can achieve what you want, with images being cross-posted on your workflow page, by using another node we implemented called "Civitai Hash Fetcher (Image Saver)". You will need this so it fetches your current workflow hash AFTER you upload your workflow, since you need to upload first to get the hash. With this in place (BEFORE you upload), the correct hash will be stored in your workflow when a user queues it.
![image](https://github.com/user-attachments/assets/3e2306ad-c845-45bf-873c-9604139836c9)
